core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -655,11 +655,13 @@ public long getNewContainerId() throws UnavailableException {
    * @return a {@link JournalEntry} representing the state of the container id generator
    */
   private JournalEntry getContainerIdJournalEntry() {
-    BlockContainerIdGeneratorEntry blockContainerIdGenerator =
-        BlockContainerIdGeneratorEntry.newBuilder().setNextContainerId(mJournaledNextContainerId)
-            .build();
-    return JournalEntry.newBuilder().setBlockContainerIdGenerator(blockContainerIdGenerator)
-        .build();
+    synchronized (mBlockContainerIdGenerator) {
```

Comment:
why is this synchronized now?

Reply:
O, I think this is strictly more correct, since `mJournaledNextContainerId` is protected by `mBlockContainerIdGenerator`.